### ISSUE TITLE
Issue With Readme when on step Stateful Components

### ISSUE THREAD

**Issue by pengcheng95** [Microsoft/TypeScript-React-Starter#199]
In the state of the component, currentEnthusiasm is declared as 
```
currentEnthusiasm: props.enthusiasmLevel || 1
```
This creates a problem later for your tests when you pass enthusiasm level as 0 to the component
```
<Hello name='Daniel' enthusiasmLevel={0} />
```
Since props.enthusiasmLevel is 0, it is falsy so currentEnthusiasm becomes 1 which makes it fail the test: throws when the enthusiasm level is 0.

The problem can be fixed if you declare currentEnthusiasm as a ternary such as 
```
currentEnthusiasm: props.enthusiasmLevel === undefined ? 1 : props.enthusiasmLevel 
```
or declare default props for the component. The ternary seems to be a better fit for this guide.

**Comment by sohara** [comment]
Agreed that there is an issue here. There's at least one other issue/PR related to this: https://github.com/Microsoft/TypeScript-React-Starter/pull/192

My solutions was a little different from yours (could be improved by using types/interface for `defaultProps` and `mergedProps`:
```ts
  constructor(props:Props) {
    super(props);
    const defaultProps = { enthusiasmLevel: 1 }
    const mergedProps = { ...defaultProps, ...props };
    this.state = { currentEnthusiasm: mergedProps.enthusiasmLevel };
  }
```
